% 서울대학교 전기공학부 (전기컴퓨터공학부) 석사ㅡ 박사 학위논문
% LaTeX 양식 샘플
\RequirePackage{fix-cm} % documentclass 이전에 넣는다.
% oneside : 단면 인쇄용
% twoside : 양면 인쇄용
% ko : 국문 논문 작성
% master : 석사
% phd : 박사
% openright : 챕터가 홀수쪽에서 시작
\documentclass[oneside,phd]{snuthesis}

\include{snutocstyle} % SNU toc style

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% 다른 패키지 로드
%% http://faq.ktug.or.kr/faq/pdflatex%B0%FAlatex%B5%BF%BD%C3%BB%E7%BF%EB
%% 필요에 따라 직접 수정 필요
\ifpdf
	\input glyphtounicode\pdfgentounicode=1 %type 1 font사용시
	%\usepackage[pdftex,unicode]{hyperref} % delete me
	\usepackage[pdftex]{graphicx}
	%\usepackage[pdftex,svgnames]{xcolor}
\else
	%\usepackage[dvipdfmx,unicode]{hyperref} % delete me
	\usepackage[dvipdfmx]{graphicx}
	%\usepackage[dvipdfmx,svgnames]{xcolor}
\fi
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\usepackage{textcomp}
\usepackage{lipsum} % lorem ipsum
\usepackage{tabularx}
\usepackage{longtable}
\usepackage{booktabs}
\usepackage{caption}
\usepackage{graphicx}
\usepackage{multirow}
%\MakeOuterQuote{"}
\usepackage[margin=.4cm]{caption}
\usepackage[font=small,labelfont=bf]{caption}

\usepackage{amsmath}
\usepackage{nccmath}
\usepackage{mathtools}

\usepackage{array}
\usepackage[table]{xcolor}
\usepackage{ragged2e}
%\usepackage{ctable}

%\usepackage{tocstyle}
%\usetocstyle{standard}
%\settocfeature{raggedhook}{\raggedright}
%\usepackage{setspace}

\newcolumntype{L}[1]{>{\raggedright\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{C}[1]{>{\centering\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{R}[1]{>{\raggedleft\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}	

\renewcommand{\figurename}{Figure}
\renewcommand{\tablename}{Table}

\usepackage{float}			% for float objects
\usepackage{subfloat}		
\usepackage{subfigure}		% for adding several figures in a figure environment
\usepackage{graphicx}

%% \title : 22pt로 나오는 큰 제목
%% \title* : 16pt로 나오는 작은 제목
\title{Informatics system for analyzing and comparing RNA-seq data using network-based gene prioritization}
\title*{RNA-seq 데이터를 분석하고 비교하기 위한 네트워크 기반 유전자우선순위분석 정보시스템}

\academicko{이학}
\schoolen{COLLEGE OF NATURAL SCIENCES}
\departmenten{INTERDISCIPLINARY PROGRAM IN BIOINFORMATICS}
\departmentko{협동과정 생물정보학}

%% 저자 이름 Author's(Your) name
%\author{홍길동}
%\author*{홍~길~동} % Insert space for Hangul name.
\author{Benjamin Hur}
\author*{허~영~회} % Same as \author.

%% 학번 Student number
\studentnumber{2012-23113}

%% 지도교수님 성함 Advisor's name
%% (?) Use Korean name for Korean professor.
%\advisor{홍길동}
%\advisor*{홍~길~동} % Insert space for Hangul name.
\advisor{Sun Kim}
\advisor*{김~선}

%% 학위 수여일 Graduation date
%% 표지에 적히는 날짜.
%% 학위 수여일이 아니라 논문 발간년도를 적어야 할 수도 있음.
%\graddate{2010~년~2~월}
\graddate{AUGUST 2019}

%% 논문 제출일 Submission date
%% (?) Use Korean date format.
\submissiondate{2019~년~5~월}

%% 논문 인준일 Approval date
%% (?) Use Korean date format.
\approvaldate{2019~년~5~월}

%% Note: 인준지의 교수님 성함은
%% 컴퓨터로 출력하지 않고, 교수님께서
%% 자필로 쓰시기도 합니다.
%% Committee members' names
\committeemembers%
{이병재}%
{김선}%
{손현석}%
{황대희}%
{김광수}
%% Length of underline
%\setlength{\committeenameunderlinelength}{7cm}

\begin{document}
\pagenumbering{Roman}
\makefrontcover
\makefrontcover
\makeapproval

\cleardoublepage
\pagenumbering{roman}
% 초록 Abstract

\keyword{Gene prioritization, Protein-Protein Interaction, Venn diagram, Network Propagation}
\begin{abstract}
\noindent
Technology advances in transcriptomics achieved to measure the expression level of genes on a large scale with high resolution. Since the number of genes in transcriptome data is large, researchers need help to identify genes that are relevant to their research. One popular way is to use the integration of heterogeneous databases (biological network, database, text-mining, and more). However, the relationship among heterogeneous databases is often too complex to understand which genes are relevant. Recently, it is shown that gene prioritization methods are effective to help researchers identify genes that are relevant to their research. In my doctoral study, I developed three bioinformatics systems that perform network-based gene prioritization analysis of transcriptome data and produce genes in ranks relevant to specific conditions.

The first study was to develop a gene prioritization system that utilizes the characteristics of RNA sequencing data. The system prioritizes genes by (i) removing less informative differentially expressed genes (DEGs) using gene regulatory network (GRN), biological pathways, and (ii) filter out genes that have single nucleotide variances (SNVs) between samples. As a result, this study was able to show that the integration of network and SNV information was able to increase the performance of gene prioritization. The key idea of the method is to reconfirm the DEGs are resulted by the effect of gene knockout rather than genetic differences of different samples.

The second study was to develop a gene prioritization system that allows the user to specify the context of the experiment. Even though gene prioritization methods succeeded to filter out less significant genes by combining multiple data and methods, the results still might not focus on the context of the experiment. Therefore, it is necessary to rank genes by considering the user's context. As a result, a web tool named CLIP-GENE (context-laid integrative analysis for gene prioritization) was developed. CLIP-GENE prioritizes genes of KO experiment by (i) removing less informative DEGs using GRN, (ii) discard genes that have sample variance with SNV, and (iii) rank genes that are related to the user's context using text-mining technique as well as considering the shortest path of protein-protein interaction (PPI) to the KO gene.

The last study was to develop an informative system that can be used to compare multiple biological experiments in Venn diagram by performing gene prioritization analysis. Venn diagram is widely used to summarize DEGs among different biological experiments. However, there are several issues when Venn diagram is used to compare multiple experiments. A major issue is that there are quite a number of genes in a section of Venn diagram. Providing a large number of DEGs to the researcher is not very helpful to design follow-up experiments. Another, more serious issue is that each experiment uses a different control and DEGs from different experiments have different meanings. Thus, it is sometimes not correct to simply count how many DEGs are common between experiments. We show that Venn diagram, coupled with gene prioritization methods, can be used to compare multiple biological experiments effectively. As a result, we developed Venn-diaNet, a Venn diagram based network propagation analysis framework that prioritizes genes by comparing a number of DEG lists from multiple experiments. We demonstrated that Venn-diaNet was able to reproduce original findings in experiments of analyzing and comparing multiple biological transcriptome data measured in multiple conditions. 

In summary, the thesis summarizes that gene prioritization methods can be more powerful when combined with biological network, text-mining, and Venn diagram to prioritize phenotypic-specific genes. For each approach, we implemented software packages and web tools to support researchers' convenient access to the methods.
\end{abstract}

\tableofcontents
\listoffigures
\listoftables

\cleardoublepage
\pagenumbering{arabic}

\chapter{Introduction}
%RNA가 많이 쓰인다, 하지만 어렵다.
RNA-seq is the most popular technology that estimates the abundance of global mRNAs to find genes that is relevant to the experimental design.
The data provides an unprecedented amount of information and details that cannot be handled in a single process. 
Therefore, the expression profile from RNA-seq data is generally analyzed by using multiple databases and methods in order to obtain useful knowledge.

%RNAseq 데이터를 분석하기 위해서 많은 방법론이 있지만, 개별적으로는 어렵다.
For more than a decade, studies have introduced a number of approaches and applications to analyze RNA-seq data.
However, it is largely difficult to pinpoint a gene with enough evidence to infer the relationship between the gene and phenotype in a single analysis.
For example, differentially expressed genes (DEGs) analysis, an analysis that finds genes that have statistically altered expressions, cannot explain why and how the phenotype is different between samples.
%개별적으로 어렵기에 조합해서 사용하기도하지만, 여전히 어렵다.
To overcome the limitation of the single-step analysis, a number of studies have introduced a strategy that combinations multiple data sources and methods to compensate for the insufficient information of a single analysis (Figure \ref{gene_prioritization_workflow}).
\begin{figure*}
\begin{center}
\includegraphics[scale=0.5]{gene_prioritization_workflow.JPG}
\end{center}
\caption{Work flow for prioritizing genes (Moreau and Tranchevent, 2012)}
\label{gene_prioritization_workflow}
\end{figure*}

However, there are a number of databases and analysis methods available for the transcriptome analysis, and the strategies of combining these elements have also been diversified (Figure \ref{gene_prioritization_methods}). 
Therefore, it is now a challenging task to find a strategy that has an appropriate combination to analyze the RNA-seq data tailored for the goal of analysis.

My doctoral study addresses the challenges in analyzing RNA-seq data with three informatics systems that prioritize genes based on networks. 
The first study is to analyze RNA-seq data that have a small number of samples. 
The second study is to overcome the knowledge gap between genes in databases. The third study is to analyze complicated RNA-seq data that have multiple conditions and treatments with intuitive interpretation.

\section{Challenges of analyzing RNA-seq data}

The challenges to find phenotype-related genes with RNA-seq data can be summarized into three reasons. 
\textit{(i)} 
The amount of database and methods are huge that cause difficulties to use the correct combination.
\textit{(ii)} 
The knowledge gap between well-studied genes and less-studied genes in databases.
\textit{(iii)} Complicated experiment designs (i.e: multiple control/treatments, a small number of samples) that is difficult to analyze.
\begin{figure*}
\begin{center}
\includegraphics[scale=0.6]{gene_prioritization_methods.PNG}
\end{center}
\caption{Various strategies to prioritize genes (Moreau and Tranchevent, 2012)}
\label{gene_prioritization_methods}
\end{figure*}
\subsection{Excessive amount of databases and analysis methods}

The intensive researches made the number and the variety of databases and methods to become very large. 
However, ironically, with the given a number of options, it became difficult to choose a certain combination that would be good for their own research.
Studies have suggested various strategies \citep{moreau2012computational}(Figure \ref{gene_prioritization_methods}) to find promising genes but the strategies contain difficulties that need to be addressed.

For example, filtering strategy, a strategy that combines multiple databases (or methods) as filters and removes less significant candidate genes step by step (Figure \ref{gene_prioritization_methods}a), is a straight forward strategy that strictly reduces the number of candidates that does not satisfy each criterion. 
However, if the filters do not have enough discrimination power, it will not screen out less promising candidates. 
On the contrary, if the filters are too stringent, the strategy will cause a number of false negatives.
Thus, the strategy is very challenging to adjust the level of discrimination power according to the combination of data sources.

Unlike filtering strategy, data fusion strategy has the advantage to avoid the false negatives caused by the hard thresholds among filters by scoring the candidates at each data sources and summarizes the overall ranks (Figure \ref{gene_prioritization_methods}b).
However, because the strategy combines heterogeneous data sources, the relationship between input and output between data sources becomes complicated, and the complexities increase according to the number of data sources.
Therefore, it is difficult to make an intuitive interpretation for the final results and this makes the analysis tools to have a `black-box' like characteristics \citep{moreau2012computational}.

Thus, despite the abundance and variety of databases and methodologies, it is still difficult to combine them effectively.


\subsection{Knowledge gap between genes in databases}

The knowledge gap between genes also causes difficulties for strategies that attempt to find promising genes with prior knowledge.
A well-studied gene will likely appear in any databases even if the researcher did not intend to consider this gene during the whole experiment design.
Thus, it is necessary to address the biased-rank genes as well as prioritizing genes that are focused on the context of experiment design.

The knowledge gap also affects the network-based strategies that use network propagation.
Network propagation is now one of the most powerful and common techniques to investigate the relationship between candidate genes and known genes (usually disease-related genes), but the strategy heavily relies on seeds that requires heavy prior knowledge to select proper seeds. 
If the prior information is not enough to select appropriate seed genes, the results of the network propagation will become less guaranteed to find the similarity between two different genes.
Of course, there are a few studies that use the sequence features or topology features instead of prior knowledge to overcome these difficulties \citep{lopez2004genome, adie2005speeding,chen2009toppgene}.
However, it is still difficult for the seed selection for transcriptome-based experiments that have poor prior knowledge.

\subsection{Complicated experiment designs}

DEGs are one of the common element that is used as initial candidates and combined with other data sources in order to find promising genes.
However, discovering phenotype-dependent gene from complicated experiment designs, such as mice gene knockout (KO) experiments that have a small number of samples, or experiments that have multiple controls/treatments, is difficult with current strategies.

For example, if the number of samples is small, DEG will have weak statistical evidence and infers that DEGs might not be caused by the phenotypic difference but the other effects, such as the genetic difference between biological replicates.
If the false positives are considered during the gene prioritization, the strategies that combine DEGs will be difficult to find true phenotypic-related genes. 

Experiment designs that compare multiple DEG list usually have samples that not only differs in its treatments but also in its controls that makes the combinations of control and treatments much more complicated.
Since each DEG list with different controls (or treatments) indicates different biological differences, simply adding or subtracting the entries between these lists is not effective to find genes that have functional interest.


\section{My approach to address the challenges for the  analysis of  RNA-seq data}

This thesis introduces three studies, each of the studies introduces an informatics system that uses a unique combination of network and data source to solve the challenges prioritizing genes that are related to the phenotypic difference.

\noindent \textbf{1. Combined analysis of gene regulatory network and SNV information enhances identification of potential gene markers in mouse knockout studies with a small number of samples :} a filtering strategy that addresses the challenge of complicated experiment design that has a small number of samples by
(i) removing less informative DEGs using gene regulatory network (GRN), biological pathways, and (ii) filter out genes that have single nucleotide variances (SNVs) between samples.
As a result, this study was able to show that the integration of network and SNV information was able to increase the performance of gene prioritization. The key idea of the method is to reconfirm the DEGs are resulted by the effect of gene KO rather than genetic differences of different samples.

\noindent \textbf{2. CLIP-GENE: a web service of the condition-specific context-laid integrative analysis for gene prioritization in mouse TF knockout experiments :} a data fusion strategy that prioritizes genes of KO experiment to address the difficulties of knowledge gap between genes by (i) removing less informative DEGs using GRN, (ii) discard genes that have sample variance with SNV, and (iii) rank genes that are related to the user's context using text-mining technique as well as considering the shortest path of protein-protein interaction (PPI) to the KO gene.

\noindent \textbf{3. Venn-diaNet : Venn diagram based network propagation analysis framework for comparing multiple biological experiments} a Venn diagram based network propagation analysis framework that prioritizes genes that address the challenge of complicated experiment design that have multiple controls and treatments as well as overcoming the knowledge gap in seed selection.
Venn-diaNet was able to reproduce original findings in experiments of analyzing and comparing multiple biological transcriptome data measured in multiple conditions. 


\section{Background}
\subsection{Differentially expressed genes}

Once the global mRNA expression of genes are measured, estimating the DEGs between samples is one of the great starting point to understand the characteristics of the phenotype differences \citep{marioni2008rna}. 
Measuring the expression differences that is statistically significant have proved to be a successful approach to find genes that is responible for the phenotypic differences \citep{hardcastle2010bayseq, robinson2010edger, anders2012differential, trapnell2013differential,leng2013ebseq,li2013finding,tarazona2015data}.
The statistical approach to calculate DEG varies based on the distributional assumptions. Softwares such as DEGseq \citep{wang2009degseq}, MyRNA \citep{langmead2010cloud}, and PoissonSeq \citep{li2012normalization} uses Poisson model for RNA-seq count data while edgeR \citep{robinson2010edger}, DESeq \citep{anders2012differential}, and DESeq2 \citep{love2014moderated} uses negative binomial model. 
In addition, there are more DEG calculation software tools that use more other statistical models. However, it is important to understand the characteristics of the models and carefully apply to the data \citep{huang2015differential}.

\subsection{Gene prioritization}
Gene prioritization is a strategy that identifies the most promising genes from a large pool of candidates by integrating multiple data source (Figure \ref{gene_prioritization_methods}). 

The integration of the list of genes and external data sources allows increasing the data dimension from 1D (simple gene list) to a higher dimension that can have much more explanation to the data \citep{moreau2012computational, cowen2017network}.
Currently, the strategy of gene prioritization can be generally categorized into four types.
\textit{(i)} filtering strategy, \textit{(ii)} profiling and data fusion, \textit{(iii)} text-mining, and \textit{(iv)} network analysis \citep{moreau2012computational}. 


Filtering strategy is a approach that uses multiple data source (or methods) as filters while each filter removes less significant candidate genes step by step (Figure \ref{gene_prioritization_methods}a).
Unlike filtering strategy, data fusion strategy has the advantage to avoid the false negatives caused by the hard thresholds among filters by scoring the candidates at each data sources and summarizes the overall rank (Figure \ref{gene_prioritization_methods}b). 
Text-mining is a data mining method that finds the associations between given keywords. 
In bioinformatics, text-mining is a strategy that finds the associations between candidate genes and knowledge (disease, phenotype or else) while the relationship between two elements is retrieved by information retrieval methods \citep{krallinger2008linking, winnenburg2008facts}(Figure \ref{gene_prioritization_methods}c).
\begin{figure*}
\begin{center}
\includegraphics[scale=0.7]{network_based.png}
\end{center}
\caption{Work flow for prioritizing genes (Moreau and Tranchevent, 2012)}
\scriptsize{Network-based strategy is to find the similarity between candidate genes and seed genes using networks (Figure \ref{gene_prioritization_methods}d, Figure \ref{network_based}) while seed genes are often defined as disease genes or phenotype-relevant genes that requires prior knowledge.}
\label{network_based}
\end{figure*}






\section{Outline of the thesis}

Chapter 2 introduces that the integration of network and SNV information was able improve the statistical bias from mice gene knockout experiments that have small number of samples.
Chapter 3 describes a system that combines GRN, PPI, and text-mining technique was able to prioritize genes that focuses the context of the experiment.
Chapter 4 suggests that Venn diagram has a great advantage to prioritize genes that can address the challenge of heterogeneous data and seed selection issues for network propagation.
Chapter 5 summarizes and concludes the studies that is introduced in this thesis.

The thesis is concluded by an appendix the bibliography of the cited references.



\chapter{A filtering strategy that combines GRN, SNV information to enhances the gene prioritization in mouse knockout studies with small number of samples}

\section{Background}

DEGs from RNA-seq data are often used for finding significant genes that can explain the phenotypic differences between control and cases \citep{oshlack2010rna, frazee2014differential}.
However, in gene knockout studies, discovering phenotype-dependent gene only with DEG can be difficult because distinguishing whether the expression alteration is resulted by the inactivation of the knockout gene or by the genetic variations that were merely from differences in samples rather than phenotypic differences.
And the problem becomes much more challenging when the number of samples is small, an issue that RNA-seq experiments face frequently \citep{tarazona2011differential}.
Various methods and models were proposed to overcome the difficulties of selecting phenotype related DEGs from a small number of samples such as the Poisson model \citep{marioni2008rna}, Bayesian approaches \citep{10.1093/nar/gkn705, anders2010differential}, or increased the sequence depth of samples \citep{tarazona2011differential}. 

Even if a number of studies have resolved the difficulties of DEG detection in some degree, addressing phenotype related DEGs from a small number of samples is still a challenging process.
Studies suggested to increase the number of biological samples is the most critical factor have significant DEGs \citep{10.1093/bioinformatics/btt688}.
However, increasing the number of biological samples is not easy for many reasons. 
Thus, a new approach that can detect significant gene markers in
a small number of samples is necessary.
This study proposes a new method that distinguishes genes that are relevant to the phenotypic differences in mice gene knockout experiments that have small number of samples.
The method uses filter-out gene prioritization strategy that combines GRN, biological pathways and SNVs information using DEGs as input \citep{hur2015combined}.


\section{Methods}
%%%%%%%%%%%FIGURE%%%%%%%%%%%
\begin{figure*}
\begin{center}
\includegraphics[scale=0.6]{combined_analysis.png}
\end{center}
\caption{Filtering strategy combining networks and SNV}
\label{combined_analysis}
\end{figure*}
%%%%%%%%%%%FIGURE%%%%%%%%%%%
The gene prioritization method uses a reductionist approach by adding more filters at each step as described below.
\begin{enumerate}
\item The first filter is to use a method to identify DEGs between control and case samples. In this study, we used fold change, a classical DEG selection method.
\item The filter at the second step is to use GRN. GRN is constructed from large volume of public data to represent the whole gene regulatory network. DEGs that are included in the network are selected as candidates.
\item The third filter utilizes biological pathway information. 
Candidates that are not included in the pathways are discarded.
\item Finally, candidates that have higher than a certain rate of SNVs are discarded since the DEGs that have SNVs possibly resulted from genetic differences rather than phenotypic differences.
\end{enumerate}

\subsection{First filter : DEG}

From the given expression profile, DEGs are considered as initial candidates.
DEGs are used for the purpose of observing the alteration of expression patterns that could explain the phenotypic differences among samples.
DEGs were selected by using fold change of the expression value (FPKM) between case and control. 
The study used multiple cutoffs in order to compare and observe differences in the number of selected genes.
Note that this study used samples that does not have enough biological replicates to perform statistical testing to calculate DEGs. Therefore, expression fold change was used as a DEG estimation


\subsection{Second filter : GRN}

The concept reverse engineering the regulatory network from transcriptome data, GRN is a very effective method that can consider complex relationships of many genes \citep{basso2005reverse}.
GRN is used as the second filter of the gene prioritization process in order to discard genes that have less significant roles in the regulatory network.

In order to construct a GRN that is appropriate for mice gene knockout experiment data, public data (Microarray, RNA-seq) of mice were collected from NCBI GEO. 
For microarray, each series matrix files from GSE45929 \citep{ramsey2013fgfr2}, GSE16741 \citep{yun2010microrna}, GSE30906 \citep{shan2012cigarette}, GSE36780 \citep{bae2012mirna}, GSE40375 (not published), GSE41380 \citep{nusinow2012network}, GSE43663 \citep{ruan2013proteoglycan} were used for GRN construction. 
These data contains gene expression value of multiple samples that differs in mouse’s strain, genotype and treatment (42 samples in total) and were created by the same microarray platform (Illumina MouseWG-6 v2.0 expression beadchip) and preprocessed by R bioconductor lumi package \citep{du2008lumi} (variance stabilizing transform, quantile normalization). 
The study integrates gene expression values of 7 series matrix files (GSE45929, GSE16741, GSE30906, GSE36780, GSE40375, GSE41380, GSE43663) into a single matrix and quantile normalized gene expression values of every sample and used it as an expression profile for construction of GRN.

GRN is constructed by using NARROMI \citep{zhang2012narromi} while a list of transcription factors and co-factors from the Animal Transcription Factor Database \citep{zhang2011animaltfdb} was used for regulator information for NARROMI.
For the gene list, we simply defined it as a list of whole genes that includes not only transcription factors and co-factors but also non-transcription factors. 
As a result, NARROMI constructed a network topology of 2950865 edges. 
The study supports a URL for the network topology file which was used in this study (\url{epigenomics.snu.ac.kr/mouse\_network/total\_mouse.topology}).

With the constructed GRN, the study discards candidates that have weak or no regulatory roles. 
The method filters out less significant DEGs that do not have any potential regulatory roles upon the calculated GRN.
As a result, candidates that participate a regulatory role remains


\subsection{Third filter : Biological Pathway}

The combination of DEG and GRN information was used not only for reducing the number of candidates but also to select significant genes that have regulatory roles that could represent the phenotypic differences between WT and KO mice. 
However, GRN is a hypothetical topology that gains regulatory information from the given data. Therefore, it is also important to ensure whether the candidates have biological evidence.
In this study, KEGG pathway \citep{kanehisa2000kegg} for confirming the candidates in terms of domain knowledge.


\subsection{Final filter : SNV}

Even if the study reduced the number of candidates by using multiple filtering methods, it is necessary to eliminate genes that have genetic differences that may not represent phenotypic differences. 
Since the statistical power is weak in a small number of samples, it is difficult to distinguish whether the genetic differences were caused by phenotypic differences or not. 
Therefore the study removed genes that have certain or higher SNV rate. 
This process will remove SNVs from the genetic differences but also by the phenotypic differences. A possibility to have false negative results.
However, it will completely avoid the risk of selecting SNVs resulting from genetic differences. 



\section{Results and Discussion}

GSE47851 were used for evaluating the filter-out method. 
The performance of the method is discussed by comparing between the genes reported from the original research article \citep{yagi2014transcription} and the genes prioritized by the filtering method. 

RNA-seq data of GSE47851 are from an experiment of Gata3 KO that have multiple SRA files. The study used 8 SRA files (SRR896215, SRR896216, SRR896217, SRR896218, SRR896219, SRR896220, SRR896221, SRR896222) that have two conditions where each of the conditions have 2 biological samples and 2 technical replicates of each biological sample.

The study reported that genes of TNF and TNFR super families, members of NFkB and cell surface markers of ILC2s have expression alterations when Gata3 is not activated in ILC2 cells \citep{yagi2014transcription}. 
The authors reported that when Gata3 is inactivated, many TNF and TNFR superfamily genes, such as Tnfrsf9 and Tnfsf21 and NFkB family members, including Nfkb2 and Relb, have altered expression patterns while cell-cycle inhibitor Cdkn2b was up regulated.
According to the authors we report, the reductionist approach was able to reproduce 4 out of 5 genes (except Tnfsf21).
In addition, we were able to reconfirm the following facts by mapping the candidate genes to the KEGG pathway. 
Figure \ref{study1_pathway1} represents expression alteration in NF-kappa B signaling pathway, showing down regulations of Nfkb2 (p100) and Relb when Gata3 is inactivated. 
Expression alteration was also detected in the TNF signaling pathway (Figure \ref{study1_pathway1}). 
TNF and TNFR super family genes, such as Tnf and Tnfrsf9, were successfully detected in the pathway as well as the statement (Figure \ref{study1_pathway2}). 

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{study1_pathway1.png}
\end{center}
\caption{Expression alteration in NF-kappa B signaling pathway \citep{hur2015combined}}
\scriptsize{
{(A) NF-kappa B signaling pathway mapped with non-filtered candidates. With no filtering method, too many genes are shown in the pathway which makes it difficult to find an appropriate gene marker.
(B) NF-kappa B signaling pathway mapped with candidates
filtered by DEG. The number of genes is greatly reduced compared to the non-filtered method. However, difficulty exists in finding significant
gene markers as the number of genes is still too great. 
(C) NF-kappa B signaling pathway mapped with full-filtered candidates. The number of
genes was greatly reduced compared to non-filtered or DEG-only filtered candidates while keeping the genes reported by Yagi et al.(2013).}}
\label{study1_pathway1}

\end{figure*}

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{study1_pathway2.png}
\end{center}
\caption{Expression alteration in TNF signaling pathway \citep{hur2015combined}}
\scriptsize{
{(A) TNF signaling pathway mapped with non-filtered candidates. With no filtering method, too many genes are shown in the pathway which makes it difficult to find an appropriate gene marker. 
(B) TNF signaling pathway mapped with candidates filtered by DEG.
Number of genes are greatly reduced than non-filtered method. However, difficulty exists in finding significant gene marker as the number of genes are still too many. 
(C) TNF signaling pathway mapped with full-filtered candidates. The number of genes were greatly reduced than nonfiltered or DEG-only filtered candidates while keeping the genes reported by Yagi et al.(2013).}}
\label{study1_pathway2}
\end{figure*}

\begin{table}
\centering
\begin{tabular}{l || c c c c c}
\hline
Filtering Steps& NONE  &  1\textsuperscript{st} filter  &  2\textsuperscript{nd} filter  &  3\textsuperscript{rd} filter & Final filter \\ \hline \hline
SelectedCandidates & 12298 & 2153  & 1184  & 478   & 343   \\ 
TruePositives      & 23    & 23    & 19    & 18    & 14    \\ \hline
Accuracy           & 0.002 & 0.827 & 0.905 & 0.962 & 0.972 \\ 
Precision          & 0.002 & 0.011 & 0.016 & 0.038 & 0.041 \\ 
Recall             & 0.885 & 0.885 & 0.731 & 0.692 & 0.538 \\ 
F-measure          & 0.004 & 0.021 & 0.032 & 0.073 & 0.076 \\ \hline
\end{tabular}
\caption{Performance comparison of filters}
\justifying{\noindent\scriptsize{The table represents the remaining candidates, number of correctly predicted true positives, and the performance of each adapted filters.}}
\label{1st_table}
\end{table}

The study also stated about the expression alterations in cell-surface markers of ILC2s. 
The study reported that 130 genes are positively regulated by GATA3 in ILC2s, and not in Th2 cells. 
Cell-surface markers of ILC2s, such as Icos, Il2ra, Kit, Il1r2, Cysltr1, Htr1b, and Tph1 were included.
As a result, the reductionist approach was able to reproduce 4 genes among 7 were successfully matched (Figure \ref{study1_pathway3}C).

In addition, the study evaluated whether each filter had a significant role during the filter-out process (Table \ref{1st_table}). 
Table \ref{1st_table} summarizes the performance of prioritizing candidates at each filtering step. 
Without no filter (NONE), it is obvious that there is a very few chance to prioritize genes reported in the original paper. 
However, when filters are gradually added, number of false positives decreased rapidly. 
In addition, the recall have steadily decreased at each filtering steps, but the F-measure represents that the general performance of the filtering process was better than the previous steps.

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{study1_pathway3.png}
\end{center}
\caption{Expression alteration in Cell cycle pathway}
\scriptsize{
(A) TNF signaling pathway mapped with non-filtered candidates. With no filtering method, too many genes are shown in the pathway which makes it difficult to find an appropriate gene marker. 
(B) TNF signaling pathway mapped with candidates filtered by DEG. Number of genes are greatly reduced than non-filtered method. However, difficulty exists in finding significant gene marker as the number of genes are still too many. 
(C) TNF signaling pathway mapped with full-filtered candidates. The number of genes were greatly reduced than non-filtered or DEG-only filtered candidates while keeping the genes reported by Yagi et al.(2013).}
\label{study1_pathway3}
\end{figure*}




\section{Discussion}

This study proposed a novel method that use a four filtering steps to distinguish phenotype-dependent genes from RNA-seq data of mouse knockout studies that have small number of samples. 
The study demonstrated that the combination of DEG, GRN, biological pathways
and SNV information was able to narrow down the significant genes that have regulatory roles and reduced the risk of including candidates that have genetic differences.
However, several limitations of this study need to be addressed.
First of all, there should be more rigorous study of GRN construction. 
Using much omics data for GRN construction somehow preserves important relationships between transcription factors and their target genes, but how much data is needed for GRN construction is not rigorously studied. 
In this study, we had enough omics data for the network construction, therefore we were able to use a simple method using NARROMI \citep{zhang2012narromi}.
However, when the amount of omics data for network construction is not enough, special techniques such as low order partial correlation based methods \citep{zuo2014biological} should be consider.
Second, removing genes with genetic variation allows us to focus on genes that are relevant to the underlying biological mechanisms for the knockout study. 
However, genetic variations do not always affect the transcription activity of genes, and it is possible that the suggested method might discard a number of SNVs that were affected by the knockout gene.
Thus, it is necessary to investigate the effect of genetic variations on transcription activities.



%%%%%%% 2nd study %%%%%%
\chapter{An integration of data-fusion and text-mining strategy to prioritize context-laid genes in mouse TF knockout experiments}

\section{Background}

To overcome the limitations of the DEG methods, studies suggested data fusion techniques that utilize additional information to effectively identify genes that are related to the phenotypic differences. 
However, it is known that the integration of heterogeneous databases have several difficulties while prioritizing candidates for data of gene knock study that motivated this study.
First, most of the existing gene prioritization tools are not appropriate for the condition specific data such as mice knockout data. 
When a certain gene is knocked out, researchers have specific hypotheses that are related to the observed phenotypic differences. 
Thus, to select genes that are related to phenotypic differences, it is important
to not only consider gene expression alteration but also to prioritize genes with the researcher’s interest. 
Without considering the condition or the goal of experiment, gene prioritization will likely to focus on genes that have enough supported evidence instead of considering the intention of the experiment design.
The best strategy is to provide information about the conditions of the experiment or specific hypothesis that the user has. 
When the user provides such information, genes can be prioritized by consulting the literature database. 
Therefore, it is necessary to perform an integrative analysis of transcriptome data and literature data for the condition specific gene selection and prioritization.

Second, complex relationships among genes should be considered in order to selected and prioritize genes that are related to the phenotype. 
Therefore, networks such as GRN and PPI are useful in explaining alteration among genes by considering gene-gene and regulatory relationships.
Many knockout experiments investigated transcription factors (TFs) that could result in
the phenotypic differences by analyzing the GRN \citep{geier2007reconstructing, madhamshettiwar2012gene, wang2012inference, ud2015optimal}.

Thus, considering GRN (to be specific, GRN) is essential to characterize the roles of TFs from knockout data. 
In addition to GRNs, PPI networks also assist in explaining expression alteration among genes since PPI networks consist more entities than other networks such as GRNs and biological pathway networks. 
Since we need to use both TF and PPI networks, an issue is how to utilize two different networks in a single computational framework. Our approach uses GRN to select candidate genes from TF knockout experiment and uses PPI to prioritize candidate genes in combination of the literature information in a condition specific manner.

Third, existing computational methods for prioritizing genes are not designed for mouse knockout data. 
Only 3 among 27 tools (listed in Gene Prioritization Portal \citep{tranchevent2010guide}) are designed for the mouse data \citep{van2012genefriends, tranchevent2008ndeavour,nitsch2011pinta}.

However, these tools are generally not applicable to evaluate RNA-seq data of knockout experiments. 
For example, even though PINTA \citep{nitsch2011pinta} and GeneFriends \citep{van2012genefriends} can prioritize genes based on the concept of the guilt-by-association or network analysis, these tools require a pre-selected gene list of a certain size: up to 200 genes in PINTA and up to 500 genes in GeneFriends.
Both tools are not applicable when the number of genes are large, such as DEG results. 
Although use of a stringent cutoff value can reduce the number of candidate genes that can be used for aforementioned tools, there may be too many false negatives.
Therefore, the requirement of a pre-selected gene list in PINTA and GeneFriends is not easy to be resolved.
In addition, PINTA is designed for microarray data and prioritizes genes by referring the expression profiles of its neighbors from the PPI network, but it does not consider the influence of the knockout gene. 
Likewise, GeneFriends prioritizes genes by considering co-expression of other
genes but does not reflect the effect of the knockout gene.
Another tool, Endeavor \citep{tranchevent2008ndeavour}, is able to prioritize genes from a large number of gene list that does not require pre-selection from gene list. 
However, Endeavor requires a gene list from prior knowledge for a training dataset, and it is designed to select disease related genes rather than knockout related genes. 
To address the discussed issues, this study developed CLIP-GENE (Context Laid Integrative analysis to Prioritize genes) \citep{hur2016clip}. 
A web based tool that takes a DEG list as input and uses GRN and SNV information to narrow down candidate genes and prioritizes genes with PPI information and literature information. 
In particular, CLIP-GENE allows researchers to specify the context of the experiment as a set of keywords input to a bio-medical entity search tool (BEST) \citep{lee2016best}.

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{clip_gene_workflow.png}
\end{center}
\caption{A Workflow of CLIP-GENE \citep{hur2016clip}}
\scriptsize{
CLIP-GENE prioritize user-interested genes that are relevant to phenotypic/functional differences of knockout mice data. CLIP-GENE takes DEG as input and filter out genes by using GRN and SNV information. Then prioritize these genes by using BEST and PPI information}
\label{fig:clip_gene_work_flow}
\end{figure*}


\section{Methods}

CLIP-GENE prioritizes genes with two major steps, selection and ranking.
For the selection step, GRN and SNV information are used to select candidate genes that are affected by the knockout gene as well as expressed differentially between wild type and knockout mice. 
For the ranking step, BEST and PPI information are used to prioritize genes according to the researcher’s context or hypothesis. 
With the assistance of a BEST \citep{lee2016best}, it allows to specify certain context or hypothesis with a set of keywords by user that is expected from the data.
Afterwards, PPI is used to consider the gene-gene relationship between the candidate genes and the knockout gene.
Workflow of CLIP-GENE is illustrated in Figure \ref{fig:clip_gene_work_flow}.
Details of each step are described below.

\subsection{Selection of initial candidate genes.}

CLIP-GENE takes a DEG list from the knockout experiment and investigates the regulatory role of the DEGs by referring to GRN. 
GRN is created using NARROMI \citep{zhang2012narromi} with data of normal inbred mice data that varied in its strains, developmental stage, and tissues (150 samples of wild type mice RNA-seq data from 17 independent studies) \citep{yao2014corepressor, tena2014comparative, stilling2014k, srivastava2015astrocyte, shen2014tet3, roger2014otx2, ntziachristos2014contrasting, moniot2014prostaglandin, mielcarek2014dysfunction, liu2014pax5, kayo2014mir, harmacek2014unique, gu2014weak, deng2014single, bhatnagar2014genetic, altboum2014digital, alpern2014taf4}.
while a list of transcription factors and co-factors from the Animal Transcription Factor Database \citep{zhang2011animaltfdb} was used for regulator information for NARROMI.

CLIP-GENE takes a list of DEGs as input and uses them as initial candidates. 
Then, by referring to the mouse GRN that was constructed using 150 mice expression profiles, DEGs that do not affect other DEGs or DEGs that are not affected by the knockout gene are excluded. 
This step is performed to focus on the relationship between the regulator and its target genes that are significantly altered.

After CLIP-GENE selects candidate DEGs that takes a part in the regulatory role, SNV information is used to filter out DEGs that might be caused by the genetic differences rather than the influence of the knockout gene.
It is well known that even if the inbred mice are raised in a controlled environment, genetic differences are likely to be present \citep{EISENERDORMAN2009318}. 
If a large number of RNA-seq experiments can be performed, it is possible to screen genes that may be expressed differentially due to the genetic difference. However, it is not practical to perform such a large number of RNA-seq experiments that is enough to remove such genes. 
To compensate the low statistical power of the typical RNA-seq data, candidate genes with over than a certain rate of SNVs in the knockout mice are
discarded \citep{hur2015combined}.

\subsection{Prioritizing genes with the user context and PPI}
Candidate genes selected in the previous step are ranked in terms of the relevance to the phenotype in two different criteria: the user specified context and the PPI information.

Researchers can specify their hypothesis for the knockout data as ‘context’ in a set of keywords. 
Specifically, context means a set of subjective words that describe the user’s
interest such as ‘expected biological function when the gene is knockout’ or ‘known function of the knockout gene’. 
For example, a context for Gata3 knockout data can be described as ‘Immune response’, ‘Cell signaling’, or ‘Inflammatory response’ \citep{yagi2014transcription,WAN2014233}. 
Then genes that are related to the user-specified keywords can be determined
by looking for the relevance between keywords since certain
keywords are documented in the literature in relation
to a certain gene. 
Thus this can be viewed as a process to find keyword-keyword relationship and keyword-gene relationship to prioritize genes.
In order to find the relevance between two different keywords, literature search systems based on the named entity recognition (NER) are known to be effective \citep{10.1093/bib/bbr018}.
For CLIP-GENE, BEST \citep{lee2016best} is used to find the relevance between knockout gene and candidate genes as well as the relationship between candidate genes and the user given context. 
With the user specified keywords, BEST computes relevance between any pair of keywords from PubMed and returns a relevance score of genes with ranks.
Once the relevance score of ‘context to candidate gene’ and ‘knockout gene to candidate gene’ is calculated, the maximum of them is used to represent how the candidate gene is relevant to the user’s interest or the knockout gene.
As a result, a candidate gene with a higher relevance score is ranked with higher priority.

PPI information is used to rank candidates by computing the shortest interaction path to the knockout gene on the STRING PPI network \citep{szklarczyk2010string}. 
Candidates that have shorter interaction path to the knockout gene are considered to be more relevant to the phenotypic/functional difference, hence
they are ranked with a higher priority. 
Finally, CLIPGENE summarizes candidates with ranks by combining
the BEST and PPI information with unweighted Borda count \citep{10.2307/227640}. 
Figure \ref{fig:clip_gene_1st_filter} and \ref{fig:clip_gene_2nd_filter} describes the overview of gene prioritization.

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{clip_gene_1st_filter.png}
\end{center}
\caption{Gene selection and ranking process \citep{hur2016clip}}
\scriptsize{Prioritizing genes with Biomedical Entity Search Tool (BEST). BEST is utilized to find the relevance between knockout gene
and candidate gene as well as the relationship between candidate gene and given context. Then CLIP-GENE retrieves the maximum score to
represent that the candidate gene is highly relevant to the user’s interest or knockout gene. As a result, candidate gene with higher relevance score
is ranked with high priority}
\label{fig:clip_gene_1st_filter}
\end{figure*}


\begin{figure*}
\begin{center}
\includegraphics[scale=0.5]{clip_gene_2nd_filter.png}
\end{center}
\caption{Gene ranking process \citep{hur2016clip}}
\scriptsize{Prioritizing genes with Biomedical Entity Search Tool (BEST) and PPI information. CLIP-GENE summarizes ranks from step 1 and PPI shortest path by using Borda count}
\label{fig:clip_gene_2nd_filter}
\end{figure*}


\section{Results and Discussion}

For the performance evaluation, we used datasets that come with publications reporting which genes are relevant to the functional difference when the gene is silenced.
These genes are used as true positives to measure the precision, recall, and F-measure in terms of genes reported in the publications for data sets, GSE47851 \citep{yagi2014transcription}, GSE54932 \citep{ZHANG20141989}, and GSE53398 \citep{zhuang2014barx2}.
CLIP-GENE was compared with methods and tools that can be used for RNAseq mouse data. 
This study compared DEG-only method (DEG), integrative analysis method (IA) \citep{hur2015combined}, and GeneFriends \citep{van2012genefriends} in terms of the predictive power.
In addition, since the user can specify context with a set of keywords, the performance depends on the context that the user provides. 
In this experiment, four different sets of keywords are used as context. 
To compare the predictive power, the study designated the best case and the worst case in terms of the number of genes reproduced by CLIP-GENE. 
In addition, as BEST investigates the relationship between two given keywords by referring
the abstract from PubMed, we chose keywords that were not mentioned in the abstract of the corresponding publications. 
This process is done to make sure that BEST did not consider the keywords from the publication
that generated the data while calculating the relevance score.

Dataset GSE47851 is from a Gata3 knockout mouse study that reported 25 genes were relevant to the functional difference between the wild type and the knockout.
For the performance evaluation, four different contexts: ‘Inflammatory response’, ‘Immune regulation’, ‘Cell differentiation’, ‘Cell proliferation’, the known functions of Gata3 \citep{yagi2014transcription,WAN2014233}.
Dataset GSE54932 is from a Setd2 knockout study, reporting 21 genes that are relevant to
the phenotypic/functional differences between the wild type and the knockout. 
‘Cell proliferation’, ‘DNAmismatch repair’, ‘Endodermal differentiation’, and ‘Histone modification’
were used as the contexts for the Setd2 knockout study since they are keywords representing well-known functions of Setd2 \citep{ZHANG20141989, Feng2015}. 
Dataset GSE53398 of Barx2 knockout mice, was used for the last evaluation. 
The study reported that 47 genes significantly differs when Barx2 is silenced. 
For the corresponding knockout mice data, we used ‘Myoblast progeny’, ‘Muscle maintenance’,  ‘Chondrogenesis’, ‘Morphogenesis’ as the contexts for CLIP-GENE \citep{OLGUIN2004375, mi2016panther, Zammit347, meech2012barx2, Meech2135, Tsau3307}.


\begin{table}
\begin{tabular}{ l c c c  }
\hline
Methods & Precision  & Recall & F-measure\\
\hline
DEG & 0.0105 & 1 & 0.0208\\
IA & 0.0239 & 0.72  & 0.0463 \\
GeneFriends & 0.0038 & 0.92 & 0.0075\\
\textbf{CLIP-GENE (Immune regulation*)} & \textbf{0.0613}  & \textbf{0.64}   & \textbf{0.1122}\\
CLIP-GENE (Inflammatory response) & 0.0354  & 0.76   & 0.0677\\
CLIP-GENE (Cell differentiation) & 0.0294  & 0.72   & 0.0564\\ 
CLIP-GENE (Cell proliferation) & 0.0201  & 0.72   &0.0391\\ \hline
\end{tabular}
\caption{Performance of CLIP-GENE while analyzing GSE47851 (Gata3 KO)}
\scriptsize{The best performed measurement is marked with a star (*) with a bold context.}
\label{clip_table1}
\end{table}



%Setd2
\begin{table}
\begin{tabular}{ l cc c  }
\hline
Methods     & Precision  & Recall   & F-measure\\
\hline
DEG & 0.0099 & 0.5238 & 0.0195\\
IA & 0.0183 & 0.1905 & 0.0333 \\
GeneFriends & 0.0015 & 0.5238 & 0.0031\\
\textbf{CLIP-GENE (Endodermal differentiation*)} & \textbf{0.2083}  & \textbf{0.2381}   & \textbf{0.2222}\\ 
CLIP-GENE (Cell proliferation) & 0.0252  & 0.3333   & 0.0468\\ 
CLIP-GENE (DNA mismatch repair) & 0.1304  & 0.1429   & 0.1364\\
CLIP-GENE (Histone modification) & 0.0408  & 0.1905   & 0.0672\\ 
\hline
\end{tabular}
\caption{Performance of CLIP-GENE while analyzing GSE54932 (Setd2 KO)}
\scriptsize{The best performed measurement is marked with a star (*) with a bold text.}
\label{clip_table2}
\end{table}

%barx2
\begin{table}
\begin{tabular}{ l cc c  }
\hline
Methods    & Precision  & Recall   & F-measure\\
\hline
DEG & 0.0071 & 0.7872 & 0.0142\\
IA &0.0111 & 0.3617  &0.0215 \\
GeneFriends & 0.0036 &  0.617 & 0.0071 \\
\textbf{CLIP-GENE (Myoblast progeny)} & \textbf{0.1818}  & \textbf{0.0426}   & \textbf{0.069}\\ 
CLIP-GENE (Muscle maintenance) & 0.0476  & 0.0426  & 0.0449\\ 
CLIP-GENE (Chondrogensis) & 0.1667 & 0.0426   & 0.0678\\
CLIP-GENE (Morphogenesis) & 0.0217  & 0.4255   & 0.0412\\
\hline
\end{tabular}
\caption{Performance of CLIP-GENE while analyzing GSE53398 (Barx2 KO)}
\scriptsize{The best performed measurement is marked with a star (*) with a bold text.}
\label{clip_table3}
\end{table}


\subsection{Performance with the best context}
In terms of F-measure, CLIP-GENE achieved better performance in finding phenotypical/functional relevant (validated) genes than other methods \ref{clip_table1},\ref{clip_table2}, \ref{clip_table3}, as well as prioritizing phenotypic/functionally relevant genes with proper ranks \citep{hur2016clip}. 

Context ‘Immune regulation’ achieved the best performance for the Gata3 knockout data, which performed about 5.4 times better than DEG, 2.4 better than IA, and 15 times better than GeneFriends while ranking 4 genes in the top 10 gene list among 25 validated genes. 
For the Setd2 knockout data, CLIP-GENE ranked 4 genes among 21 validated genes in top 10 list with the context ‘Endodermal differentiation’, achieving 11 times better than DEG, 6.7 times better than IA, and 72 times better than GeneFriends. For the Barx2 knockout data, context ‘Myoblast progeny’ achieved the best performance, achieving 4.8 times better than the DEG, 3.2 times better than IA method, and 9.7 times better than Gene Friends. 
In addition, CLIP-GENE was able to prioritize 2 genes among 47 validated genes in top 10 from Barx2 knockout data.

\subsection{Performance with the worst context}

In terms of F-measure, even with the worst performed context, CLIP-GENE achieved better performance in predicting phenotypic/functionally relevant genes. 
For the Gata3 knockout data, context ‘Cell proliferation’ performed 1.9 times better than DEG and 5.2 times better than GeneFriends, and slightly poor than IA. 
CLIP-GENE ranked one gene in the top 10 among 25 validated genes.
The context ‘Cell proliferation’ performed the worst case for the Setd2 knockout data, which still performed better than DEG, IA, and GeneFriends while reporting one
gene among 21 validated genes in top 10. ‘Morphogensis’ was the worst context for the Barx2 knockout dataset.
However, CLIP-GENE still performs better than other methods while ranking 2 genes from the 47 validated genes in top 10, which again suggests that CLIP-GENE promises significant results than other compared methods even with the worst context.

\section{Discussion}

The performance of CLIP-GENE depends on the context that the user provided. 
However, in terms of candidate selection and prioritization, even with the context that performed worst, CLIP-GENE was consistently superior to DEG, IA, and GeneFriends.

Transcriptome data from mouse models with certain genes knocked out are widely used to investigate gene functions in terms of phenotypes. 
In order to determine genes that are affected by the knocked out TF, both selecting
candidate genes and prioritizing genes are necessary.
Only three tools are available for the mouse data, but none of these tools was appropriate to prioritize genes of user’s interest from knockout data. 
This study presents a novel web service that select and prioritize the candidate
genes in terms of the user’s experimental context. 
Two major contributions are: (\textit{i}) CLIP-GENE allows researchers to specify the
experimental conditions in a set of keywords. 
Our system automatically determines relevance between the keywords and genes so that we can provide rankings of the candidate genes in the users' context.
(\textit{ii}) CLIP-GENE provides a comprehensive web service for the mouse knockout experiments by integrating multiple resources into a single framework: mouse GRN, SNV information, PPI network, and literature information.





%Third study
\chapter{Integrating Venn diagram to the network-based strategy for comparing multiple biological experiments}

\section{Background}

Before performing advanced analysis (i.e. network analysis, gene set analysis, or more) in transcriptome data, identifying DEGs is the very first step to understand the characteristics of the experiment. 
Since the number of DEGs can be hundreds or thousands, understanding the difference between samples with a list (or lists) of DEGs is not easy.
An effective method to summarize the large number of DEGs is to use Venn diagram.
A simple, yet a powerful tool that can illustrate the portion of each gene sets. 
The intuitive diagram helps researchers to understand the common and distinctive characteristics of the experiments that assists the decision for further investigation.
However, there are several issues when Venn diagram tries to compare and analyze multiple experiments.

First of all, current Venn diagram tools are difficult to find genes that are responsible for the phenotype differences. 
Most of the current Venn diagram applications are developed with the purpose of visualizing the correct appearance of the diagram or to compare gene sets that aid researchers' brief understanding by giving additional knowledge such as enriched sub-network or gene sets \citep{kestler2004generalized, martin2012vennture,kestler2008vennmaster, chen2011venndiagram, heberle2015interactivenn, hulsen2008biovenn, wang2014netvenn, jeggari2018evinet}.
The provided information may be useful but it is difficult to design a follow-up experiments with a simple list of gene sets.

Moreover, elucidating the phenotypic difference for the experiment designs that have different controls is also an issue.
For example, when a dataset of two experiments that focus to find the differences of gene knockout (KO) effect between liver and muscle, the DEG of each experiments represents tissue-specific phenotypic difference.
Thus, comparing the gene sets and the number of genes of the two experiments is not informative enough to pin point whether the genes are effected by the gene knock out effect or the tissue effect.

If it is possible to rank DEGs in a region of Venn diagram, then the researcher can make more informed decision and overcome the difficulties that is described. 
To rank DEGs, this study combined the gene prioritization method into Venn diagram.
Gene prioritization is a widely used method to rank genes by combining multiple database and methods to maximize the biological relevance to answer difficult question that cannot be easily solved in a single data.
Network propagation is one of the widely used technique that computes the influence of initial nodes (or seeds) to other nodes \citep{cowen2017network}, and prioritize genes  in the context of biological networks \citep{li2010genome, smedley2014walking, kohler2008walking, vanunu2010associating, lee2011prioritizing, chen2009disease, chen2006mining}. 
However, selection of seed genes is one of the critical factor for the network propagation analysis and becomes more important when prior knowledge is not available or is not enough. 
This paper suggests that the seed selection issue can be handled by allowing the user to select seed genes freely in arbitrary combinations of regions in a Venn diagram. We present Venn-diaNet: a web-based Venn diagram based network analysis framework that can prioritize genes to compare multiple biological experiments of transcriptome data.
A convenient web-based user interface is provided to generate Venn diagrams of DEGs dynamically and to perform network propagation experiments to investigate which genes are relevant to certain phenotypes.
This study suggests that Venn diagram, coupled with analytic methods such as network propagation, can be a very useful tool for comparing multiple biological experiments with different controls.


\section{Methods}

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{Venndianet_workflow.png}
\end{center}
\caption{Venn-diaNet work flow}
\label{workflow}
\scriptsize{Step 1 : Venn-diaNet receives DEG lists per experiments from user.
Step 2 : Uploaded DEGs from step 1 are interpreted with a Venn diagram as well as organized as sets with table.
Step 3 : Define specific or multiple C$_i$ as seeds for further network propagation analysis.
Step 4 : Once seed is defined, Venn-diaNet instantiates a PPI network of DEGs from STRING DB. Network propagation with given seeds from the previous steps. As a result, DEGs are ranked by the probability score calculated during the Markov Random Walk.}
\end{figure*}

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{venndianet_propagation.png}
\end{center}
\caption{Key concept of Venn-diaNet}
\label{propagation}
\scriptsize{(A) Instantiate a PPI network with the DEGs from the multiple experiments.
(B) When we are interested in C$_{1}$ that has similar function as C$_{2}$, we can define C$_{2}$ as seeds.
(C) Performing network propagation with Markov Random Walk.
(D) Discard C$_{3}$ genes (as well as seed genes) in order to focus on C$_{1}$ genes. Remaining genes are ranked by the probability score calculated from the previous step.}
\end{figure*}

\subsection{Taking input data}
Venn-diaNet takes multiple DEG lists as input while each DEG list is resulted by the comparison of treatment/control or treatment/treatment experiment (Figure \ref{workflow}: Step 1). 
Each file must include one DEG list from one experiment.
For example, if a researcher wants to compare three different experiments, three independent files of DEG list must be provided. 
The format of the file is as follows. 
Each input file requires gene ID (transcript ID) for the first column and gene symbol for the second column. 
We provide an example data on the web page of Venn-diaNet for better understanding.

\subsection{Generating Venn diagram of DEG sets}
Venn-diaNet considers each experiment as a set for the diagram. 
Therefore, With given number (=$n$) of experiments $E$, Venn-diaNet generates a diagram of n circles that have $2^n-1$ regions. 
Each region is denoted as $C_i$ ($1 \le ≤ i \le 2^n-1$) while each $C_i$ contains genes of 

%%%%%%%% EQUATION %%%%%%%%
\begin{ceqn} 
\textbf{C}_i &= \{ \textbf{g} :  \textbf{g} \in \bigcap_{j=1}^{N} \textbf{G} (\textbf{b}_j) \} 
\end{ceqn}
, \hspace{5mm}
\begin{ceqn}
    \textbf{G($b_j$)} = \begin{cases}
        E_j & if \enspace j = 1 \\
        E^c_j & if \enspace j = 0
    \end{cases}
\end{ceqn}
%%%%%%%% EQUATION %%%%%%%%

$b$ represents the binary number of $C_i$ (i.e. $C_1$ = 001) while $b_j$ indicates the position of digits (i.e. $b_1$ = 1, $b_2$ = 0, $b_3$ = 0).
If Venn-diaNet receives DEG lists from 3 experiments, Venn-diaNet illustrates a Venn diagram of 3 sets ($E_1$,$E_2$,$E_3$) that have 7 regions ($C_1$,$C_2$,$C_3$, $\cdots$ $C_7$), where
$C_7$ contains genes of $E_1 \cap E_2 \cap E_3$. 
$C_i$ represents specific DEGs to certain region that could be considered as `condition specific genes’.

\subsubsection{Seed selection}
This step is the most important part of Venn-diaNet. 
A user can select multiple (or a single) $C_i$ as seeds for network propagation to measure the global influence of the seed DEGs.
Thus, the results will vary depending on the selected seeds. 
Network propagation methods generally use informative genes as seeds. Such as `disease-related genes', `phenotype-related genes', or else.
The idea of network propagation in Venn-diaNet is very similar but does not need to select genes that require prior knowledge.
As the DEG in each region of the Venn diagram can be considered as condition-specific DEGs, the DEG in $C_i$ can be a guide to find the similarities or dissimilarities to other $C_j$ ($j \ne i$) that researchers are interested in.
Because the selection is crucial, this study provides three possible seed selection scenarios to help understanding the seed selection.


The first scenario is to consider \textit{`condition-specific function'} as seeds. 
Again, DEGs in specific region can be considered as condition-specific DEGs.
If researcher use these genes as seeds, it can prioritize DEGs belonging to other conditions in terms of functional similarity to the seed DEGs. 
For example, if a user wants to prioritize tissue A-specific DEGs (Figure \ref{propagation}A: $C_{1}$) that have similar function to the tissue B-specific DEGs when the same genes is knockout (KO), tissue B specific-DEGs (Figure \ref{propagation}A: $C_{2}$) can be used as seeds.

The second scenario is to consider \textit{`common function'} as seeds. 
In some cases, a user might be interested in condition specific DEGs that have common function in different experiments.
For instance, if the user is interested in tissue A-specific DEGs (Figure \ref{propagation}A: $C_{1}$) that have similar function between two different tissues, $C_{3}$ can be seeds. 
Similarly, if the common KO effect in different tissues are in interest ($C_{3}$), $C_{1}$+$C_{2}$ can be seeds. 

The last scenario is to consider seeds that have \textit{`Functional similarity'}. 
Distinct from the two scenarios stated above, this study assumed a case that there is no sufficient knowledge to select certain condition as seeds. 
In this case, a `minimum guideline' to choose certain conditions as seeds to rank the genes of interest.
If the user have multiple experiments and expects some DEGs in condition of interest ($C_{i}$) to have functional similarity to other condition DEGs ($C_{j}$), the condition that have functional similarity to the condition of interest will be appropriate to be as seeds. 
This guideline is suggested to prioritize genes for experiments that studies compound effects of multiple treatments which will be introduced later.


\subsection{Network propagation and gene ranking}

When a set of seed DEGs are selected, Venn-diaNet instantiates a protein-protein interaction (PPI) network of DEGs from STRING DB \citep{szklarczyk2014string}.
In the instantiated network, nodes are DEGs and an edge between two DEGs is defined when the corresponding edge in the original PPI network is of high-confidence (combined score $>$ 700).
Then, Markov Random Walk (MRW) \citep{diffusr} is performed using the seeds selected in the previous step (Figure \ref{workflow}: Step 4). 
The goal of network propagation is to quantify the influence of seed DEGs to the remaining DEGs. 
The selected seed DEGs can be considered as the hypothesis that a user wants to test. 
Thus, by performing a network propagation analysis, the user can obtain the DEGs pertaining to the hypothesis. 
For the network propagation, an R package {\tt diffusr}, the implementation of MRW, is used. The equation is shown below:

\begin{ceqn}
\begin{align*}
p^{t+1}=(1-r)A'p^t+rp^0
\end{align*}
\end{ceqn}

where $p^0$ is the vector of initialized nodes, $t$ is a time step, $p^t$ is the vector at the current time step $t$, $p^{t+1}$ is the vector at the next time step, $A'$ is column-normalized matrix of adjacency matrix $A$, and $r$ is the restart rate.
$p^0$ is initialized in 1 or 0, to represent the assigned seed DEGs and target DEGs, and normalized so the sum of the elements in $p^0$ becomes 1. 
The adjacency matrix $A$ is a matrix consists with 0 or 1 that represents a graph with no weighted edges. 
0.5 is used for $r$ and network propagation stops when $L1$ norm difference between $p^{t}$ and $p^{t+1}$ is smaller than $10^{-4}$, which are the default progress of the {\tt diffusr} package. 
When the algorithm stops, Venn-diaNet returns a ranked gene sets based on the network propagation result.

\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{WEB_workflow.png}
\end{center}
\caption{Venn-diaNet (web) work flow}
\label{web_workflow}
\scriptsize{A work flow of Venn-diaNet (web).
Step 1: Upload DEG list per experiment.
Step 2: Select seed condition $C_i$ 
Step 3: Perform analysis.
Venn-diaNet gives user (1) list of ranked genes, (2) gene's neighbor nodes information (when the node is clicked). (3) Venn diagram with PPI network (when the Venn diagram is zoomed in).}
\end{figure*}

\section{Results and Discussion}

This study evaluated the performance of Venn-diaNet using three datasets downloaded from the Gene Expression Omnibus (GEO) \citep{edgar2002gene} or from the supplementary data of the corresponding published paper. 
Three datasets were selected to show how to perform network propagation analysis with different seed gene selections.

%FIGURE%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{PER2_GOTERM_COMPARISON.png}
\end{center}
\caption{Venn-diaNet Per2 GO term Comparison}
\scriptsize{(A) Venn-diagram of GSE20165 experiment. $C_1$ represents Per2 KO vs WT DEGs that is specific to BAT while $C_2$ represents WAT specific Per2 KO vs WT DEGs.
(B) Enriched GO terms by DAVID gene functional clustering analysis. Gene functional clustering was performed for each specific condition $(C_i)$.
(C) Enriched GO terms of Top 30 genes prioritized by corresponding seeds.}
\label{result_figure_per2}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%%%%%
\begin{table}[]
\centering
\begin{tabular}{l|| c c c c }
\hline
Gene & FC  & $C_1$  & $C_3$  & $C_1$+$C_3$ \\ \hline
Ucp1 & 2   & 18  & 16  & 14    \\
Cidea  & 4   & 26  & 18  & 25    \\
Acsm3  & 47  & 30  & 39  & 35    \\
Pdk4   & 20  & 71  & 61  & 74    \\
Cpt1b   & 11  & 6   & 20  & 6     \\
Acads    & 129 & 58  & 27  & 61    \\
Acadm    & 119 & 14  & 15  & 11    \\
Acadl    & 95  & 52  & 28  & 58    \\
Acadvl    & 67  & 37  & 12  & 34    \\
Hadha   & 111 & 5   & 10  & 3     \\
Hadhb    & 54  & 8   & 13  & 5     \\
Cox7a1  & 14  & 62  & 66  & 67    \\
Cox8b   & 12  & 22  & 43  & 28    \\ \hline
PredictedCandidates & 120 & 100 & 100 & 100  \\ \hline
\end{tabular}
\caption{Comparing ranking results of the Per2 KO experiment performed by BenedettoGrimaldi}
\label{venndianet_table1}
\end{table}

%%%%%%%%%%% CASE 1 %%%%%%%%%%%%%%%%%
\subsection{Venn-diaNet for two experiments}

The dataset is from a study of Per2 KO mice with two different tissues: (\textit{i}) Per2 KO vs WT in white adipose tissue (WAT Per2 KO), and (\textit{ii}) Per2 KO vs WT in brown adipose tissue (BAT Per2 KO).
The authors used these DEGs and reported that several WAT specific expressed genes have similar behavior also BAT when Per2 is KO.

Venn-diaNet used these two experiments from this study to evaluate how well Venn-diaNet could reproduce the effects of the corresponding study. 
For convenience, this study denoted BAT Per2 KO specific DEGs as $C_1$, WAT Per2 KO as $C_2$, and the intersection DEGs of BAT Per2 KO and WAT Per2 KO as $C_3$ (Figure \ref{result_figure_per2}A).
Venn-diaNet used this data to show that Venn-diaNet can reproduce the authors results by following the authors inputs, interest, and approach. 
The original paper reported that Per2 KO caused BAT specific genes to express in WAT by controlling PPAR$\gamma$-dependent genes.
Therefore, the aim of this study is to find promising $C_2$ DEGs that have the similar characteristic in BAT tissue.
Three suggested seed scenarios can be used to address the authors interests.
For each seed scenarios, the study compared (\textit{i}) how the GO terms of ranked top 10\% genes matches the GO terms reported in the original paper, and (\textit{ii}) 
how many genes matches to the genes that were reported in the original paper.
Note that the authors used only fold change to rank genes and did not use any gene prioritization method.

\subsubsection{Condition specific function ($C_1$) \& common function as seeds ($C_3$)} 
BAT Per2 KO specific DEGs ($C_1$), can be used as seeds in order to prioritize genes of WAT Per2 KO specific DEGs ($C_2$). 
This scenario is to investigate that some of the unknown PPAR$\gamma$-dependent genes that expresses exclusively in BAT somehow seems to be expressed in WAT when Per2 is KO.
The phenomenon indicates that there might be functional similarity between these two different conditions. Likewise, common DEGs between two experiments ($C_3$) can also be considered as seeds. 
Activation of BAT-specific PPAR$\gamma$-dependent genes in WAT also means that BAT and WAT have common functions. Thus, the common function of these genes ($C_3$) might be guideline  to prioritize WAT specific genes with the context of `functional similarity' between two different tissues.
It is interesting that Venn-diaNet could prioritize genes in top 30 (about 10\% of total candidates) as well as prioritizing genes that are related to the functions that the authors reported (Figure \ref{result_figure_per2}C, Table \ref{venndianet_table1}).

\subsubsection{Analysis scenario with functional similarity as seeds ($C_1$)} 

As discussed in the previous section, researchers might encounter a situation where the user does not have sufficient knowledge to select seeds. 
In this case, the suggested `minimum guideline' to choose certain condition as seeds to rank genes in condition of interest.
For this, the study defined `The condition that have functional similarity to the condition of interest will be appropriate to be as seeds' as a `minimum guideline' to find seeds. 

The process is very straight-forward. 
(\textit{i}) Find the major GO terms of each $C_i$, 
and (\textit{ii}) use genes in $C_i$ if the GO terms are similar to the condition $C_j$ ($j \ne i$) that we want to prioritize.
As a result, the study found that GO term (mitochondrion) in $C_1$ was similar to the condition of interest ($C_2$) (Figure \ref{result_figure_per2}B). Thus, $C_1$ becomes appropriate seed for this scenario and the results shares the same which we discussed in the previous subsection.

Venn-diaNet is also tested with other possible seed scenario ($C_1 + C_3$) to confirm whether Venn-diaNet performs better than random seeds.


\begin{table}[]
\begin{tabular}{l || c c c c c c c c} \hline
Gene    & FC  & $C_2$  & $C_4$  & $C_6$  & $C_2$+$C_4$ & $C_2$+$C_6$ & $C_4$+$C_6$ & $C_2$+$C_4$+$C_6$ \\\hline
Ccl3    & 284 & 236 & 27  & 159 & 78    & 246   & 35    & 91       \\
Ccl6    & 107 & 15  & 188 & 171 & 53    & 24    & 206   & 62       \\
Ccl28   & 257 & 36  & 220 & 242 & 94    & 54    & 240   & 107      \\
Cd14    & 39  & 174 & 21  & 107 & 44    & 186   & 25    & 51       \\
Cxcl1   & 12  & 125 & 131 & 142 & 144   & 143   & 148   & 156      \\
Cxcl2   & 5   & 132 & 9   & 139 & 17    & 151   & 11    & 16       \\
Cxcl3   & 9   & 139 & 166 & 202 & 232   & 161   & 184   & 248      \\
Cxcl5   & 1   & 179 & 43  & 196 & 63    & 202   & 52    & 74       \\
Cxcl16  & 268 & 121 & 139 & 129 & 159   & 141   & 156   & 169      \\
Ecm1    & 207 & 238 & 312 & 282 & 320   & 258   & 319   & 324      \\
Enpp3   & 346 & 14  & 230 & 122 & 71    & 21    & 241   & 77       \\
Il1a    & 45  & 232 & 179 & 118 & 269   & 242   & 191   & 278      \\
Il1b    & 111 & 117 & 34  & 20  & 32    & 114   & 19    & 21       \\
Il1f6   & 213 & 378 & 378 & 373 & 385   & 380   & 380   & 387      \\
Il23a   & 389 & 62  & 177 & 164 & 88    & 73    & 192   & 101      \\
Il33    & 104 & 364 & 104 & 358 & 208   & 366   & 123   & 223      \\
Met     & 211 & 127 & 60  & 21  & 52    & 118   & 27    & 34       \\
Pglyrp1 & 16  & 73  & 341 & 303 & 179   & 86    & 347   & 199      \\
Pycard  & 226 & 50  & 241 & 172 & 115   & 60    & 250   & 134      \\
S100a8  & 7   & 248 & 65  & 121 & 130   & 257   & 77    & 143      \\
S100a9  & 3   & 227 & 238 & 39  & 296   & 188   & 218   & 286      \\
Spp1    & 99  & 80  & 189 & 155 & 118   & 92    & 205   & 135     \\\hline
\end{tabular}
\caption{Comparing ranking results of the E6/E7 experiment performed by Megan et al.}
\label{venndianet_table2}
\end{table}


%FIGURE%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{E6E7_GOTERM_COMPARISON.png}
\end{center}
\caption{Venn-diaNet HPV experiment GO term Comparison}
\scriptsize{(A) Venn-diagram of the experiment by (Spurgeonet al., 2017).
$C_1$ $C_2$, and $C_4$ represents E6/E7+E2 specific DEGs, E6/E7 specific DEGs, and E2 specific DEGs, respectively.
(B) Enriched GO terms by DAVID gene functional clustering analysis. Gene functional clustering was performed for each specific condition $(C_i)$.
(C) Enriched GO terms of Top 100 genes prioritized by corresponding seeds.}
\label{result_figure_e6e7}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%% CASE 2 %%%%%%%%%%%%%%%%%

\subsection{Venn-diaNet for three experiments}
Data from a study of human papillomavirus oncogenes \citep{spurgeon2017human} is used for Venn-diaNet validation to consider the case of more complicated experiment designs.
The study observes the independent, synergistic effects of two treatments: (\textit{i}) K14E6/E7 bitransgenic mice vs WT mice (E6/E7), (\textit{ii}) estrogen treated mice vs WT mice (E2), and (\textit{iii}) K14E6/E7 bitransgenic mice treated with estrogen mice vs WT mice (E6/E7+E2) (Figure \ref{result_figure_e6e7}).

The study focused on E6E7+E2 DEGs ($C_1$ + $C_3$ + $C_5$ + $C_7$) to determine the synergistic effect of E6/E7 and E2.
E6/E7 specific DEGs and E2 specific DEGs ($C_2$ + $C_4$) were selected for the seed scenario of `condition specific function'.
The seed scenario represents that the independent effect of each treatment as a guideline to find the effect of the combined factors.
The goal for this experiment is to reproduce GO terms and genes that the authors reported. 

\subsubsection{Condition specific function as seeds ($C_2$ + $C_4$)} 
As a result, Venn-diaNet could prioritize genes and GO terms that were reported in the original paper by using the combination of independent effects of two factors as seeds ($C_2$ + $C_4$) (Figure \ref{result_figure_e6e7}C and \ref{venndianet_table2}). 
However, several careful consideration remains to be discussed. 
When Venn-dianet considers the prioritized top 20\% genes, Venn-diaNet was not superior than the authors approach, but it could prioritize genes that are related to the GO terms were the original paper focused. 
In addition, Venn-diaNet could prioritize other genes that were related to the function of interest (immune response \& inflammatory response) that are responsible to the HPV associated cervical cancer while the authors did not.

For example, Tlr2, a gene that is known to be related to take a significant
role in HPV associated cervical cancer \citep{woodby2016interaction, zom2016tlr2, halec2018toll, yang2018relationship}, was distinctively over expressed in E6/E7+E2. The results supports that Tlr2 might be one of the significant gene that is enhanced by the combined effect of E6/E7 and E2, which achieves the condition of `inflammatory response are increased by epithelial E6/E7 expression and further enhanced by estrogen'.
The study conjectured that Tlr2 was not included in the original paper because the fold change of Tlr2 is not significant (ranked 332\textsuperscript{th} in terms of fold change rankings). 
However, our gene prioritization analysis ranked Tlr2 much higher in the 33\textsuperscript{rd} place.

Likewise, CD74 has been reported that it may play an important role in the pathogenesis and angiogenesis of cervical cancer \citep{cheng2011expression} as well as the influence of the HPV \citep{klymenko2017rnaseq}. Venn-diaNet placed this gene in the 76\textsuperscript{th} position while fold change could only rank them as 182\textsuperscript{th}.
Icam1 was ranked 76\textsuperscript{th} in foldchange but had the 3\textsuperscript{rd} position in Venn-diaNet which also might have a E6/E7+E2 specific expression while Icam1 was also  reported to have a role with HPV related cervical carcinoma \citep{viac1992epidermotropism}
The comparison of Top 100 ranked genes related to `inflammatory response' \& `immune response' is summarized in \ref{venndianet_table2}.


\subsubsection{Functional similarity as seeds ($C_4$)} 

$C_4$ was selected by following the `minimum guideline' to select seeds.
Unlike `Condition specific function as seeds', seeds chosen by functional similarity performed weaker than the previous seeds. 
This is probably because the seed scenario does not reflect the effect of E6/E7.
E6/E7 is well known to change the activity of cytokine and chemokine, and Venn-diaNet could not prioritize those genes with not considering those effects in seeds (Figure \ref{result_figure_e6e7}C). 
The study emphasized that this seed scenario reflects that using seed genes from a singular treatment is not effective to rank genes that is under the influence of multiple treatments.
However, Venn-diaNet could still prioritize 7 genes in top 100 with seeds of `functional similarity' (\ref{venndianet_table2}).
In addition, Venn-diaNet also tested every other possible seeds, and the results indicates other seeds are less effective than the suggested seed scenarios.


\subsection{Venn-diaNet for eight experiments}

Case 3 uses a dataset from a study that designed the experiments with three treatments and four tissues: (\textit{i}) narciclasine (ncls), (\textit{ii}) high-fat diet (HFD), (\textit{iii}) normal chow diet (NCD), (\textit{iv}) WAT, (\textit{v}) BAT, (\textit{vi}) liver, and (\textit{vii}) muscle.
The initial number of sets of this study were extremely complicated that makes almost impossible to interpret the DEG list at once. 
Thus, the authors used a step-by-step filtering method to find promising genes for these multi-conditioned data. 
The authors searched the relation between treatments and tissues using hierarchical clustering and narrowed down to compare two DEG lists (HFD-ncls/HFD-veh, NCD-veh/HFD-veh) of muscle. 
The study reported genes that have low expression level in HFD, changes to have a high expression level when ncls was given. 
The results indicate that a natural compound ncls can attenuate diet-induced obesity and the associated genes can enhance the energy expenditure. 

To reproduce the results what the authors made, we planned two different scenarios. 
The first scenario is to follow the story of the authors: using two DEG lists. 
The authors compared the expression profile of treatments and tissues using hierarchical clustering as a very first step. 
They discovered that muscle had partial mutual exclusive expression pattern to other tissues, and made a hypothesis of `ncls might accelerate genes to be expressed again while the genes were suppressed in HFD environment in muscle'.
The study assumed to reached this step and use Venn-diaNet for the DEGs of HFD-ncls/HFD-veh and NCD-veh/HFD-veh.
Venn-diaNet will mimic this story with the concept of {\tt `Case 1: Venn-diaNet for two experiments'} analysis of Venn-diaNet.

Another scenario is to find promising genes purely by Venn-diaNet, using eight DEG lists. 
The goal of this scenario is to check whether Venn-diaNet can track down the reported genes, with a reasonable story.


\begin{table}
\centering
\begin{tabular}{l || cccc} 
\hline
     & & 2 DEG list && 8 DEG list             \\
      & FC         & $C_1$+ $C_2$      & $C_2$ & $C_3$+$C_5$+$C_{192}$ \\
\hline
Actc1 & 30.5       & 1          & 7  & -          \\
Tnni1 & 8          & 6          & 26 & 12         \\
Myl2  & 10         & 23         & 25 & 9          \\
Myh7  & 5.5        & 28         & 31 & 11         \\
Tnnt1 & 12.5       & 5          & 28 & 10         \\
Myl3  & 10.5       & 4          & 3  & 5          \\
Tnnc1 & 8.5        & 20         & 24 & 15        \\ \hline
\end{tabular}
\caption{Comparing ranking results of the HFD experiment performed by (Sofi G. Julien et al.)}
\label{venndianet_table3}
\end{table}

\subsubsection{Authors' approach : two DEG list} 
As described in the previous section, the study also assumed to performed hierarchical clustering and focus to find certain genes in C$_{3}$ (Figure \ref{result_figure_HFD}A) that have the common characteristics of up-regulation when ncls is induced  and up-regulated in NCD without any treatments . 

In order to prioritize genes in C$_{3}$, the study used the seed scenario of {\tt Condition specific function as seeds}.
DEGs that are common in both experiments can be prioritized using the independent effects of each factor.
Therefore, C$_{1}$+C$_{2}$, the independent effect of each treatments was selected as seeds to observe the influence to the genes that have same activity alteration in HFD-ncls/HFD-veh and NCD-veh/HFD-veh (C$_{3}$).
The study found that Venn-diaNet could prioritize and reproduce the genes where the authors reported (Table \ref{venndianet_table3}) as well as prioritizing GO terms of the authors' interest with better hit ratio (Figure \ref{result_figure_HFD}C). 
The minimum guideline, `Functional similarity as seeds' (C$_{2}$) showed weaker gene prioritization but still had a better focus on GO terms (Figure \ref{result_figure_HFD}C and Table \ref{venndianet_table3}).
In addition, this study is designed to find the common effect from independent conditions, meaning that the condition of interest is closely related to each other condition. 
Therefore, it is natural to have poor performance with the same reason that is discussed in the previous section.
%FIGURE%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{HFD_GOTERM_COMPARISON.png}
\end{center}
\caption{Venn-diaNet HFD GO term Comparison}
\scriptsize{(A) Venn-diagram of GSE63268 experiment. $C_1$ represents HFD (ncls/veh) specific DEGs while $C_2$ shows veh (NCD/HFD) specific DEGs.
(B) Enriched GO terms by DAVID gene functional clustering analysis. Gene functional clustering was performed for each specific region.
(C) Enriched GO terms of Top 100 genes prioritized by corresponding seeds}
\label{result_figure_HFD}
\end{figure*}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure*}
\begin{center}
\includegraphics[width=\textwidth]{HFD_story2_part1.png}
\end{center}
\caption
{Venn-diaNet using 8 different DEG list}
{
\scriptsize{(A) Using up and down-regulated DEG list to Venn-diaNet (web). The Venn diagram directly shows muscle DEGs in HFD-ncls/HFD-veh, and NCD-veh/HFD-veh are similar to each other while other tissues are not similar to each other.
(B) Using up-regulated DEG list to Venn-diaNet. The Venn diagram shows that up-regulated muscle DEGs in HFD-ncls/HFD-veh, and NCD-veh/HFD-veh are very similar to each other while other tissues are not similar to each other.}
}
\label{HFD_story2}
\end{figure*}

\subsubsection{Venn-diaNet approach: All (eight) DEG list} 
The study assumed that the researcher do not have enough knowledge to the corresponding data, and try whether Venn-diaNet could reach to the authors' conclusion. 
Venn-diaNet simplely with all DEG lists (that contains up and down-regulation) from eight different experiments at once (Figure \ref{HFD_story2}A). 
The Venn diagram shows that the intersection of HFD-ncls/HFD-veh and NCD-veh/HFD-veh shared many DEGs in muscle ($C_{48}$) than any other tissues ($C_3$, $C_{12}$, $C_{192}$).

The findings of Venn diagram reaffirms the authors' hierarchical clustering results and leads to the idea that the intersection of HFD-ncls/HFD-veh and NCD-veh/HFD-veh in muscle have common functions than other tissues, and needs to be analyzed in detail. 
To start the detailed search, up-regulated DEG list is used to examine whether Venn-diaNet can answer for the hypothesis of `ncls might accelerate genes to be expressed again while the genes were suppressed in HFD environment in muscle'. 
As a result, This study discovered that the condition of interest was much more distinct to other conditions (Figure \ref{HFD_story2}B: $C_{48}$) and the portion of common genes between HFD-ncls/HFD-veh and NCD-veh/HFD-veh in muscle was bigger than any other tissue ($C_{48}$, $C_3$, $C_{12}$, $C_{192}$).
The findings indicate that up-regulation of $C_{48}$ is likely to be more specific and distinct to other tissues. 
To prioritize genes in $C_{48}$, `common functions as seeds' is choosed for the seed scenario. 
This study selected the intersection of HFD-ncls/HFD-veh and NCD-veh/HFD-veh of other tissues as seeds ($C_3$, $C_{12}$, $C_{192}$) to represent that the function of 'ncls might accelerate genes to be expressed again while the genes were suppressed in HFD environment' in other tissues can assist to prioritize genes in muscle. 
As a result, this study was able to reproduce genes that the authors reported in their original paper (Table \ref{venndianet_table3}).

In addition to seed selection, the minimum guideline cannot be used for this complex condition data. 
The data is composed of 255 conditions that makes it is difficult to compare and analyze the GO terms of all these conditions.


\section{Discussion}
This study presented Venn-diaNet, a web-based software that does not require any additional installment or registration. 
Venn-diaNet draws a Venn diagram from a given input and prioritizes genes by network propagation.
This study suggested that a Venn diagram can support selecting seeds for network propagation and introduced several examples to show the idea can effectively prioritize genes that are related to the function of interests.
Venn-diaNet is designed not only to avoid the `black-box' issue in gene prioritization which is caused by the integration of heterogeneous databases but also to address a logical approach for seed selection of network propagation.
Venn-diaNet supports gene list with ranking and additional features that explains how the specific gene is influential to other genes. 
Venn-diaNet is available at: \url{biohealth.snu.ac.kr/software/venndianet}
Source code can be reviewed at: \url{github.com/hurben/VenndiaNet}

\chapter{Conclusion}

Identifying promising genes from a large pool of candidates that represent the phenotypic differences is one of the common goals from a variety of transcriptome analysis methods.
However, even if a number of studies demonstrated various approaches to prioritize genes, it is still difficult to decide a certain combination of analysis methods that overcomes the challenges caused by the knowledge gap between genes and complicated experiment design. This thesis summarized three studies to address these difficulties.
\begin{enumerate}
    \item A filtering strategy that combines DEG, GRN, pathways, and SNVs to handle the statistical bias caused by a small number of samples in mice gene KO data.
    \item A data fusion strategy that combines text-mining and PPI network to rank genes filtered by DEG, GRN, and SNVs.
    \item A network strategy that uses network propagation to rank candidate genes and use Venn diagram for seed selection.
\end{enumerate}
The first study analyzes RNA-seq data by (i) removing less informative DEGs using GRN, biological pathways, and (ii) filter-out genes that have SNVs between samples. 
This study was able to show that the integration of multiple filters were able to increase the performance of gene prioritization and refined the candidates by avoiding the genetic differences of different samples.

The second study developed an informatics system that allows the user to specify the context of the experiment to address the knowledge gap that prioritizes well-studied candidates instead of CLIP-GENE prioritizes genes of KO experiment by (i) removing less informative DEGs using GRN, (ii) discard genes that have sample variance with SNV, and (iii) rank genes that are related to the user's context using text-mining technique as well as considering the shortest path of PPI to the KO gene.

The last study addressed the seed selection issue by integrating Venn diagram to the network-based strategy.
The study developed an informative gene prioritization system that can compare multiple biological experiments in Venn diagram and select seed genes that is free from the pressure of prior knowledge.
The study demonstrated that Venn-diaNet was able to reproduce original findings in experiments of analyzing and comparing multiple biological transcriptome data measured in multiple conditions. 

In conclusion, this thesis summarizes the difficulties of RNA-seq analysis methods and addressed them by combining multiple methods to prioritize phenotypic-specific genes. For each approach, we implemented software packages and web tools to participate in researchers’ convenient access to the methods.



\section*{Acknowledgement}
In the beginning, I would like to express my deepest appreciation to my advisor, Prof. Sun Kim.
I cannot imagine myself without his dedicated guidance throughout my graduate study.
His advice and encouragement were always an important guiding lights towards my personal and professional development.
I can only hope that during the past several years I have been able to absorb some of his insights and intuition in bioinformatics.
OTHER TEXT WILL BE WRITTEN HERE
%Also, I would like to thank (이병재, 손현석, 황대희, 김광수 WILL TYPE PROPERLY) for the critical advice during the thesis review. The advice was very valuable and was a good impetus to develop myself. It has been seven years since I started to learn bioinformatics in this laboratory. During that time I was able to learn and inspired by a wonderful set of people. I am fortunate to study and cooperate with these lovely colleagues. 채희준, 정인욱, 안홍렬, 조겨리, 문지환 did not had a single hesitate to help me understanding bioinformatics. I will never forget their generosity and the big smiles of them. 임상수, 이상선, 강동원, my respectful colleagues, friends. I cannot describe how much I thank you. Not only because we were such a good team to accomplish our projects, but also I truly appreciate your sincere advice that made me, myself to step forward. 김민수, 오민식, 이성민, 박진우, 서석준, my mind blowing colleagues. I was always marveled at your in-depth knowledge and the passion to learn new knowledge. I was able to learn a lot and enjoyed having discussion with you. 강혜진, 임애란, 안용주, I will never forget the days we spent in the same building. I enjoyed having discussion and jokes that we had. Other members, 김인영, 이태헌, 박성준, 이도훈, 육예진수, 박민우, 박은화, 정다빈, 성인영, 이예브게니, 강민지, 김정현, 이궁 선생,


\bibliographystyle{natbib}
\bibliography{myReference}


\keywordalt{서울대학교, 전기공학부, 졸업논문}
\begin{abstractalt}
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.

서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
서울대학교 전기공학부 졸업논문 예제 파일입니다.
\end{abstractalt}

\acknowledgement
Thanks!

\end{document}

